src/utils.rs

```diff
@@ -15,3 +18,18 @@ where
 {
     empty::<T>().collect()
 }
+
+/// Union two BTreeMaps, call f to resolve conflicts if duplicate keys are encountered.
+pub fn union_b_tree_maps_with<K: Clone + Ord, V: Clone, F: Fn(&V, &V) -> V>(
```

Comment:
[nits] this is really a nit, but the std library calls its BTreeMap modele `btree_map`
```suggestion
pub fn union_btree_maps_with<K: Clone + Ord, V: Clone, F: Fn(&V, &V) -> V>(
```